src/main/java/main/internet/shop/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package main.internet.shop.security.impl;
+
+import java.util.NoSuchElementException;
+import main.internet.shop.exception.AuthenticationException;
+import main.internet.shop.lib.Inject;
+import main.internet.shop.lib.Service;
+import main.internet.shop.model.User;
+import main.internet.shop.security.AuthenticationService;
+import main.internet.shop.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user;
+        try {
+            user = userService.getByLogin(login);
+        } catch (NoSuchElementException e) {
+            throw new AuthenticationException("There are no user with such login!");
+        }
+        if (user.getPassword().equals(password)) {
```

Comment:
let's check all conditions inside `if` so that we could throw our Exception if anything goes wrong
change your userService.getByLogin() to return `Optional<User>`